src/main/kotlin/us/timinc/mc/cobblemon/spawnnotification/util/PlayerUtil.kt

```diff
@@ -8,10 +9,19 @@ import net.minecraft.world.dimension.DimensionType
 import kotlin.math.sqrt
 
 object PlayerUtil {
+    private lateinit var config: SpawnNotificationConfig
+
     fun getPlayersInRange(pos: BlockPos, range: Int, dimensionKey: RegistryKey<DimensionType>): List<ServerPlayerEntity> {
         val serverInstance = server() ?: return emptyList()
         val allPlayers = serverInstance.playerManager.playerList
 
-        return allPlayers.filter { sqrt(pos.getSquaredDistance(it.pos)) <= range && dimensionKey == it.world.dimensionKey}//.sortedBy { sqrt(pos.getSquaredDistance(it.pos)) }
+        var playersInRange = allPlayers.filter { sqrt(pos.getSquaredDistance(it.pos)) <= range && dimensionKey == it.world.dimensionKey}
+        playersInRange = playersInRange.sortedBy { sqrt(pos.getSquaredDistance(it.pos)) }
+
+        if (config.playerLimit > 0) {
```

Comment:
Need to test this in multiplayer to verify it works as intended before considering merging.